src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -276,8 +288,18 @@ public Page<Concept> findConcepts(ConceptCriteria criteria, PageRequest pageRequ
 
 	@Override
 	public void preCommitCompletion(Commit commit) throws IllegalStateException {
-		if (commit.getCommitType().equals(CommitType.CONTENT)) {
-			cachedBranchesQuery = null;
+		if (cachedBranchCriteria != null) {
+			if (BranchMetadataHelper.isCreatingCodeSystemVersion(commit)) {
+				cachedBranchCriteria = null;
+			} else {
+				for (BranchCriteria branchCriterion : cachedBranchCriteria.getBranchCriteria()) {
+					String branchPath = branchCriterion.getBranchPath();
+					if (branchPath.equals(commit.getBranch().getPath())) {
+						// Commit made on branch in cached criteria - clear criteria cache
+						cachedBranchCriteria = null;
```

Comment:
@RichardAWood We should add break when the first versioned branch is found.

Reply:
Break added